src/index.c

```diff
@@ -1640,3 +1647,243 @@ const char *IndexIterator_GetTypeString(const IndexIterator *it) {
+  if (PROFILE_VERBOSE) RedisModule_ReplyWithDouble(ctx, cpuTime);
+  if (printFull) {
+    for (int i = 0; i < ui->norig; i++) {
+      printIteratorProfile(ctx, ui->origits[i], 0, 0, depth + 1, limited);
+    }
+  } else {
+    RedisModule_ReplyWithPrintf(ctx, "The number of iterators in union is %d", ui->norig);
+  }
+}
+
+PRINT_PROFILE_FUNC_SIGN(printIntersectIt) {
+  IntersectIterator *ii = (IntersectIterator *)root;
+  RedisModule_ReplyWithArray(ctx, ii->num + 2 + PROFILE_VERBOSE);
+  RedisModule_ReplyWithSimpleString(ctx, "Intersect iterator");
+  RedisModule_ReplyWithLongLong(ctx, counter);
+  if (PROFILE_VERBOSE) RedisModule_ReplyWithDouble(ctx, cpuTime);
+  for (int i = 0; i < ii->num; i++) {
+    if (ii->its[i]) {
+      printIteratorProfile(ctx, ii->its[i], 0, 0, depth + 1, limited);
+    } else {
+      RedisModule_ReplyWithNull(ctx);
+    }
+  }
+}
+
+#define PRINT_PROFILE_SINGLE(name, struct, text, hasChild)                          \
+PRINT_PROFILE_FUNC_SIGN(name) {                                                     \
+  int verbose = PROFILE_VERBOSE;                                                    \
+  int addChild = hasChild && ((struct *)root)->child;                               \
+  RedisModule_ReplyWithArray(ctx, 2 + verbose + addChild);                          \
+  RedisModule_ReplyWithSimpleString(ctx, text);                                     \
+  RedisModule_ReplyWithLongLong(ctx, counter);                                      \
+  if (verbose)                                                                      \
+    RedisModule_ReplyWithLongDouble(ctx, cpuTime);                                  \
+  if (addChild)                                                                     \
+    printIteratorProfile(ctx, ((struct *)root)->child, 0, 0, depth + 1, limited);   \
+}
+
+typedef struct {
+  IndexIterator base;
+  IndexIterator *child;
+} DummyIterator;
+
+PRINT_PROFILE_SINGLE(printNotIt, NotIterator, "Not iterator", 1);
+PRINT_PROFILE_SINGLE(printOptionalIt, OptionalIterator, "Optional iterator", 1);
+PRINT_PROFILE_SINGLE(printWildcardIt, DummyIterator, "Wildcard iterator", 0);
+PRINT_PROFILE_SINGLE(printEmptyIt, DummyIterator, "Empty iterator", 0);
+
+PRINT_PROFILE_FUNC_SIGN(printProfileIt) {
+  ProfileIterator *pi = (ProfileIterator *)root;
+  printIteratorProfile(ctx, pi->child, pi->counter, (double)pi->cpuTime / CLOCKS_PER_MILLISEC, depth, limited);
+}
+
+void printIteratorProfile(RedisModuleCtx *ctx, IndexIterator *root, size_t counter,
+                          double cpuTime, int depth, int limited) {
+  if (root == NULL) return;
+
+  // protect against limit of 7 reply layers
+  if (depth == REDIS_ARRAY_LIMIT && !isFeatureSupported(NO_REPLY_DEPTH_LIMIT)) {
+    RedisModule_ReplyWithNull(ctx);
+    return;
+  }
+
+  switch (root->type) {
+    // Reader
+    case (READ_ITERATOR):       { printReadIt(ctx, root, counter, cpuTime);                       break; }
```

Comment:
why (READ_ITERATOR)?

Reply:
It is the iterator that actually does the reading from the inverted index

Reply:
I mean, why `(READ_ITERATOR)` and not `READ_ITERATOR`: why the brackets?